### PR TITLE
Fix copy panktee to work with new languages

### PR DESCRIPTION
**Pull request by AkalUstat**
also changes up the code to be a little simpler inside one, smaller function

### CODE REVIEW

**Comment by AkalUstat** [comment]
closes #892

**Comment by AkalUstat** [comment]
@maneetpaul @tsingh777 this is ready to be reviewed

File: www/js/copy.js [hunk 1]
```diff
@@ -100,73 +25,95 @@ function remapCeremony(unmapped) {
...
+
+  if (copyTranslation) {
+    if (isCeremony && panktee.Gurmukhi === null) {
+      toBeCopied += `\n\n${stripHTML(panktee.English)}`;
+    } else {
+      toBeCopied += `\n\n${panktee[`${translationLang}`]}`;
+    }
   }
-  if (copyPunjabiTranslation) {
+  if (copyTeeka) {
```
Comment (tsingh777) [inline www/js/copy.js]: add a null check for `panktee.Punjabi`

File: www/js/copy.js [hunk 2]
```diff
@@ -100,73 +25,95 @@ function remapCeremony(unmapped) {
...
+    store.getUserPref('slide-layout.fields.display-translation') &&
+    !(panktee[`${translationLang}`] === null);
+  const copyTeeka =
+    store.getUserPref('slide-layout.fields.display-teeka') &&
+    !(panktee[`${translationLang}`] === null);
+  const copyTranslit =
+    store.getUserPref('slide-layout.fields.display-transliteration') && !(panktee.Punjabi === null);
+
+  // start asigning the props from Object based on if they are to be copied
+  let toBeCopied;
```
Comment (tsingh777) [inline www/js/copy.js]: initialize this to empty string

File: www/js/copy.js [hunk 3]
```diff
@@ -100,73 +25,95 @@ function remapCeremony(unmapped) {
...
+    result.forEach(row => {
+      const remappedRow = global.controller.remapLine(row);
+      pankteeArray.push(remappedRow);
     });
-  } else if (idType === 'ceremony') {
-    baniDB.loadCeremony(id).then(rows => {
-      remapCeremony(rows);
+  } else {
+    isCeremony = idType === 'ceremony';
+    result = isCeremony ? baniDB.loadCeremony(id) : baniDB.loadBani(id, baniLengthMap[baniLength]);
```
Comment (tsingh777) [inline www/js/copy.js]: There's seems to be an off by 1. 

when mangal position is current saroops, sukhmani sahib is off by 1 line.

Comment (AkalUstat) [reply www/js/copy.js]: yea im not sure about that. It only happens in sukhmani sahib.

Comment (AkalUstat) [reply www/js/copy.js]: problem seems to be that its getting the preset mangal position, not the user prefered version.

Comment (AkalUstat) [reply www/js/copy.js]: fixed

File: www/js/copy.js [hunk 4]
```diff
@@ -8,90 +8,21 @@ const search = require('./search');
...
-// defualt these to true, so that we only need to change them if they are turned off
-let copyEngTranslation = true;
-let copyPunjabiTranslation = true;
-let copyTranslit = true;
-
 const baniLength = store.getUserPref('toolbar.gurbani.bani-length');
+const mangalPosition = store.get('userPrefs.toolbar.gurbani.mangal-position');
+let mangalPosString = '';
+
+if (mangalPosition === 'above') mangalPosString = 'above';
```
Comment (tsingh777) [inline www/js/copy.js]: prefer if `if` blocks used brackets.

**Pull request merged**